src/main/scala/mesosphere/marathon/api/akkahttp/v2/AppsController.scala

```diff
@@ -322,7 +332,36 @@ class AppsController(
     }
   }
 
-  private def listRunningTasks(appId: PathId)(implicit identity: Identity): Route = ???
+  private def listRunningTasks(appId: PathId)(implicit identity: Identity): Route = {
+    groupManager.app(appId) match {
+      case Some(app) =>
+        authorized(ViewRunSpec, app).apply {
+          val tasksF = async {
+            val instancesBySpec = await(instanceTracker.instancesBySpec)
+            await(runningTasks(Set(appId), instancesBySpec).runWith(Sink.seq)).map(_.toRaml)
+          }
+          onSuccess(tasksF) { tasks =>
+            complete(raml.EnrichedTasksList(tasks))
+          }
+        }
+      case None =>
+        reject(Rejections.EntityNotFound.noApp(appId))
+    }
+  }
+
+  private def runningTasks(appIds: Set[PathId], instancesBySpec: InstancesBySpec): Source[EnrichedTask, NotUsed] = {
+    Source(appIds).mapAsync(1) { appId =>
```

Comment:
this code is very similar to the tasks controller ... it would really make so much sense to extract this to some service. I am not saying it should be done in this PR...

Reply:
I am not sure about it. Code duplication is quite often a lesser evil.

Reply:
Actually we can extract such logic to a service level, e.g. `TaskService`, but I think this goes beyond the scope of the current PR.